align_system/algorithms/outlines_regression_adm_comparative.py

```diff
@@ -413,21 +443,37 @@ def top_level_choose_action(self,
         # to a single dictionary with values that are a list of samples
         predicted_kdma_values = merge_samples(predictions)
 
-        # Regress best choice
-        if distribution_matching == 'average':
+        # Select best choice
+        
+        # If we have a scalar value target, use average distribution matching
+        if target_kdmas[0].value is not None:
             # Averages over predicted score samples and selects choice with minimum MSE to target
-            selected_choice, first_justification = self.average_distribution_matching(predicted_kdma_values, target_kdmas)
+            selected_choice = self.average_scalar_matching(predicted_kdma_values, target_kdmas)
             # Currently returning the reasoning associated with the first sample for the selected choice
+        
+        # Else is we have KDE targets, use distribution_matching param to select choice
+        elif hasattr(target_kdmas[0], 'kdes') and target_kdmas[0].kdes is not None:
+            if distribution_matching == 'sample':
+                # set scalar value targets to a KDE sample
+                for kdma_idx in range(len(target_kdmas)):
+                    target_kde = kde_utils.load_kde(target_kdmas[kdma_idx], kde_norm)
+                    target_kdmas[kdma_idx]['value'] = target_kde.sample(1)
```

Comment:
Nit: If I'm understanding your overall comment on the PR correctly, this is just a random sample (not a mean, median, etc). Is it weighted by the kde or anything? Can a comment be made clarifying that? (I struggled to find the `sample` method, apologies if that answers the question)

Reply:
Yeah this is just random, I'll add that to the comment. We can add more options for `distribution_matching` down the line though, mean isn't well defined for KDE but maybe something like CDF(0.5)

Reply:
Added https://github.com/ITM-Kitware/align-system/pull/82/commits/dfadd402dc323ce4116b00fb5f99ae0900841f7b